File: app/src/main/java/beans/prj/newsapp/di/AppComponent.java
package beans.prj.newsapp.di;

import beans.prj.newsapp.NewsApplication;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

import javax.inject.Singleton;

@Singleton
@Component ( modules = {AndroidSupportInjectionModule.class
        , AppModule.class
        , ActivityBuilder.class} )
public interface AppComponent extends AndroidInjector<NewsApplication> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<NewsApplication> {}
}

File: app/src/main/java/beans/prj/newsapp/di/AppComponent.kt
package beans.prj.newsapp.di

import beans.prj.newsapp.NewsApplication
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class,
    NetworkModule::class,
    ActivityBuilder::class,
    FragmentBuilder::class]
)
interface AppComponent : AndroidInjector<NewsApplication> {

    @Component.Factory
    abstract class Factory : AndroidInjector.Factory<NewsApplication>
}